assets/js/base/hooks/cart/use-store-cart.js

```diff
@@ -35,21 +38,40 @@ const defaultCartData = {
  * @return {StoreCart} Object containing cart data.
  */
 export const useStoreCart = ( options = { shouldSelect: true } ) => {
+	const { isEditor } = useCartContext();
```

Comment:
So this could conflict with the usage of `isEditor` passed through from `useCheckoutContext` for where `useStoreCart` is used to populate checkout.  Which makes me wonder, should we introduce a _generic_ `EditorContextProvider` that generically exposes things for editor context for _any_ of our block logic that needs various data in the editor context?

Initially it just exposes `isEditor` via `useEditorContext`, but potentially it could expose other things too (`isPreview` for example)? 

If you agree, then I think it'd be better to refactor the CartContextProvider you added in this pull to be `EditorContextProvider` exposing the `isEditor` flag for now. In a subsequent pull (or here if you want) we could update things using the `isEditor` flag from the checkout context provider to use the new `EditorContextProvider` instead (and remove `isEditor` from `CheckoutContextProvider`?

As a bonus, simply wrapping components in `EditorContextProvider` would automatically set `isEditor` to `true` for the children.

Reply:
I'm fine with that. Will implement this tomorrow.